src/lambda_functions/get_prices_lambda/scraper.py

```
@@ -365,398 +388,539 @@
+            temp_path.replace(cache_path)
+            logger.info(f"Successfully cached data for: {url}")
+            
         except Exception as e:
-            logger.warning(f"Failed to save cache for {url}: {str(e)}")
-    
+            logger.error(f"Error saving to cache: {e}")
+            if temp_path.exists():
+                try:
+                    temp_path.unlink()
+                except Exception:
+                    pass
```

Ruff (0.8.2)

Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(Exception)`

(SIM105)